### PR TITLE
add resnet 20/32/44/56 variants

### PR DESCRIPTION
**Pull request by brettkoonce**
this all works, but if anybody has any suggestions on how to refactor it I would love to hear them!

### CODE REVIEW

**Comment by brettkoonce** [comment]
@rxwei over here i can refactor the inner layers slightly to be:

res20: 1/1/1  -->  1 + 1 + 1
res32: 1/3/1  -->  1 + 2 + 1
res44: 1/5/1  -->  1 + 2 + 2 + 1
res56: 1/7/1  -->  1 + 2 + 2 + 2  + 1

Would that help?  I agree that we need an array based approach long term.

File: CIFAR/resnet.swift [hunk 1]
```diff
@@ -65,26 +65,270 @@ struct BasicBlock: Layer {
...
+    init(
+        featureCounts: (Int, Int, Int, Int),
+        kernelSize: Int = 3,
+        strides: (Int, Int) = (2, 2)
+    ) {
+        self.layer1 = Conv2DBatchNorm(
+            filterShape: (kernelSize, kernelSize, featureCounts.0, featureCounts.1),
+            strides: strides,
+            padding: .same)
+
```
Comment (rxwei) [inline CIFAR/resnet.swift]: The empty lines in between these layer initializations are not very helpful and do not increase clarity. Could you remove all of them?

File: CIFAR/resnet.swift [hunk 2]
```diff
@@ -65,26 +65,270 @@ struct BasicBlock: Layer {
...
```
Comment (rxwei) [inline CIFAR/resnet.swift]: Same here.

File: CIFAR/resnet.swift (hunk 1 above)
Comment (brettkoonce) [reply CIFAR/resnet.swift]: tweaked!

File: CIFAR/resnet.swift (hunk 2 above)
Comment (brettkoonce) [reply CIFAR/resnet.swift]: tweaked!

**Pull request merged**